build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -208,19 +210,55 @@ private BuildView(Run<?, ?> build, boolean isPipeline, RelativeLocation parentJo
         this.systemTime = systemTime;
     }
 
-    private interface Reader {
-        Iterable<String> readUsersFrom(AbstractBuild<?, ?> jenkinsBuild);
+    private static Set<String> getCulpritsForWorkflowRun(WorkflowRun from, WorkflowRun to) {
+        Set<String> culprits = new TreeSet<String>();
+        WorkflowRun next = null;
+        while (true) {
+            next = next == null ? from.getNextBuild() : next.getNextBuild();
+            if (next == null || next.getNumber() >= to.getNumber()) {
+                break;
+            }
+            Iterables.addAll(culprits, getCommitters(next));
+        }
+        return culprits;
     }
 
-    private Set<String> getUsers(Reader reader) {
-        Set<String> users = new TreeSet<String>();
-
+    private static Set<String> getCommitters(Run<?, ?> build) {
+        Set<String> committers = new TreeSet<String>();
         if (build instanceof AbstractBuild<?, ?>) {
-            AbstractBuild<?, ?> jenkinsBuild = (AbstractBuild<?, ?>) build;
+            Iterable<String> iterable = transform(nonNullIterable(((AbstractBuild<?, ?>) build).getChangeSet()), new Function<ChangeLogSet.Entry, String>() {
+                @Override
+                public String apply(ChangeLogSet.Entry entry) {
+                    return entry.getAuthor().getFullName();
+                }
+            });
+            Iterables.addAll(committers, iterable);
+        } else if (build instanceof WorkflowRun) {
```

Comment:
Looks like we have two cases here, so perhaps we could have two classes providing the list of committers? One for the regular builds, one for the workflows?